runtime/src/main/java/org/corfudb/runtime/exceptions/TrimmedException.java

```diff
@@ -1,10 +1,22 @@
 package org.corfudb.runtime.exceptions;
 
+import lombok.Getter;
+import lombok.Setter;
+import org.corfudb.runtime.view.Address;
+
 /**
  * This exception is thrown when a client tries to read an address
  * that has been trimmed.
  */
 public class TrimmedException extends LogUnitException {
+    @Getter
+    @Setter
+    long address = Address.NON_ADDRESS;
+
+    @Getter
+    @Setter
+    long snapshot = Address.NON_ADDRESS;
```

Comment:
This snapshot value is populated from checkpointSnapshotAddress as the exception is propagated upwards.

Just to recapitulate, a non-recoverable exception happens when the TransactionSnapshotTime is less than the checkpointSnapshotAddress after the first retry (because the checkpointSnapshotAddress is not guaranteed to point to the latest checkpoint on the first access). Do you agree?

Those two values come from two places, one, AbstractQueuedStreamView and AbstractTransactionalContext, how would you determine whether a Trimmed Exception is non-recoverable from BackpointerStreamView if you don't have access to the retry counter?